experimental/parser/testdata/lexer/braces/stray-open.proto.stderr.txt

```diff
@@ -0,0 +1,7 @@
+error: encountered unmatched `[` delimiter
+  --> testdata/lexer/braces/stray-open.proto:2:5
+   | 
+ 2 |     [
+   |     ^ expected a closing `]`
```

Comment:
Based on the above test cases, I'm surprised this one doesn't have a `closed by this instead` pointing to the close curly brace on line 3. Is that omission intentional?

Reply:
It is. If you look at the tokens.tsv, you'll notice that this `[` is not actually closed by the `}`, and is instead a single leaf token. This allows it to be treated as an unrecognized punctuation token.

In a future enhancement, it might make sense to actually make this into a tree token that has a synthetic end token at the next newline. Recovering broken braces is more art than science and I kinda wanna keep this relatively simple for now.